File: test/cases/record/test_record_for_loop.c
#include <stdio.h>
typedef struct {
    int value;
    int step;
} counter;
counter c;

int i;


int main(void) {
    c.value = 0;
    c.step = 5;
    for (i = 1; i <= 10; i++) {
        c.value = (c.value + c.step);
        printf("%d", c.value);
    }

    return 0;
}

File: test/cases/record/test_record_for_loop.pas
program test;
type
  counter = record
    value: integer;
    step: integer
  end;
var
  c: counter;
  i: integer;

begin
  c.value := 0;
  c.step := 5;

  for i := 1 to 10 do
  begin
    c.value := c.value + c.step;
    write(c.value)
  end
end.
